src/main/scala/cognite/spark/v1/AssetHierarchyBuilder.scala

```diff
@@ -73,12 +73,12 @@ class AssetHierarchyBuilder(config: RelationConfig)(val sqlContext: SQLContext)
 
   def delete(data: DataFrame): Unit =
     data.foreachPartition((rows: Iterator[Row]) => {
-      val deletes = rows.map(r => fromRow[DeleteItem](r))
+      val deletes = rows.map(r => fromRow[DeleteItem](r)).flatMap(_.id)
       Stream
         .fromIterator[IO](deletes, chunkSize = batchSize)
         .chunks
         .parEvalMapUnordered(config.parallelismPerPartition) { chunk =>
-          client.assets.deleteByIds(chunk.map(_.id).toVector, recursive = true, ignoreUnknownIds = true)
+          client.assets.deleteByIds(chunk.toVector, recursive = true, ignoreUnknownIds = true)
```

Comment:
this is also missing support for the externalId field

Reply:
right I add support for it in the sdk => https://github.com/cognitedata/cognite-sdk-scala/pull/442/commits/805d605d5f96abc95aab2dcd2852b27a6fa0c92f :)